\documentclass[twoside]{article}
\usepackage[margin=20mm]{geometry}
\usepackage{fancyhdr}
\usepackage{mathtools,amsmath,amssymb,amsthm,mathrsfs}
\usepackage[utf8]{inputenc}
\usepackage{lmodern}
\usepackage{tikz}
\usetikzlibrary{graphs}
\usetikzlibrary{calc}
\usetikzlibrary{graphdrawing}
\usepackage{wrapfig}
\usetikzlibrary{quotes}
\usegdlibrary{layered}
\usetikzlibrary{arrows}
\pagestyle{fancy}
\rhead{Assignment 1}
\lhead{Dilawar Singh}

\title{Assignment 1}
\author{Dilawar Singh}
\date{\today}

\begin{document}

Molecules are shown in black, all reactions/processes are in blue. A label "x/y"
mean that molecule "x" is in region/place/cell "y". Arrow \tikz \draw[-|, very
thick] (0,0) -- (0.5,0);  means inhibition and  arrow \tikz \draw[-*, very
thick] (0,0) -- (0.5,0); mean facilitation of connected process. Arrow \tikz
\graph { {a}->[thick] X[blue] -> c}; means that $a$ is input to process $X$ and
$c$ is output of process $X$.

\section{The secretion of brain dilps is controlled by diet/amino acids}

\begin{wrapfigure}{r}{0.5\textwidth}
\begin{tikzpicture}[scale=1
    , every node/.style={}
    ]

    \graph[ layered layout,  nodes = {  } 
        , edges = { inner sep = 1pt, thick }
        ]
    {
        "Insulin Producing Cells" -> secretion[blue] -> "Dilps/IPC"
        -> circulation[blue] -> "Dilps/hemolymph";

        feeding[blue] -> nutrients;
        %starvation[blue] ->[-*] nutrients;
        nutrients ->[-|] circulation;

    };
\end{tikzpicture}    
\end{wrapfigure}

Dilps accumulates in IPCs when larva is starved which is reverted by refeeding
(too fast (15 minutes) to be controlled by increasing gene expression).
Starvation of amino acids is sufficient to have this response.  \footnote{Does
    lack of "feeding" increases the "secretion" with or without affecting the
    rate of circulation?}

\section{The brain IPCs couple neurosecretion with nutritional inputs}

\begin{wrapfigure}{r}{0.5\textwidth}
    \centering
    \begin{tikzpicture}[scale=1 , every node/.style={} ]
        \graph [ layered layout, nodes = { }, edge = { thick } ]
        {
            "release from IPC"[blue]; 
            secretion[blue];

            nutrients -> "???"[blue] -> secretion;
            
            "hyperpolarization/Kir2.1" [blue] ->[-|] "release from IPC";

            secretion [ blue] -> "Dilps/IPC" -> "release from IPC" -> "Dilps/hemolymph"
                -> growth [blue];

            TOR ->[-*] "release from IPC";

            %% {[same layer] TOR, "release from IPC" };
        };

    \end{tikzpicture}    
\end{wrapfigure}

Kir2.1 is a potassium channel. Overexpression of Dilps in IPC have very little
incrementally effect on groth ( $\implies$ Dilps level in IPC did not change
much ). Membrane potential is important for release of Dilps from cell body.  

\section{The fat body remotely controls insulin release from brain IPCs}


\footnote{ Don't know if ??? downregulates TOR, IIS and growth in parallel or
    some other topology exists e.g. ???? downregulates TOR which in turn
    downregulates IIS etc. }

\begin{wrapfigure}{l}{0.4\textwidth}
\begin{center}

    \begin{tikzpicture}[scale=1
        , every node/.style={}
        ]
        
        \graph[ layered layout, nodes = {}, edges = { thick } ]
        {

            "slif expr knockdown"[blue] ->[-|] "amino acids" 
            -> "????"[blue] ->[-|] { IIS, growth }; 

            "amino acids" -> TOR ->[-*] "release from IPC"[blue];
        };

    \end{tikzpicture}    
\end{center}
\caption{}
\label{fig:}
\end{wrapfigure}

\section{Direct humoral link between the fat body and the brain}

Adding 100 millimolar solution of KCl raises the Nernst potential of $K^+$ from
-80 $mV$ to -0.01 $mV$. This effectively raises the resting potential of cell to
-45 $mV$. The Nernst potential of $Na^+$ and $Cl^-$ does not change much because
of already high extracellular concentration of Chloride ions (125 millimolar)
outside. 

\begin{wrapfigure}{l}{0.4\textwidth}
    \centering
    \begin{tikzpicture}[scale=1
        , every node/.style={}
        ]

        \graph [ nodes = {} , edges = { thick } , layered layout ]
        {
            release[blue];

            "Dilps/IPC" -> release -> "Dilps/hemolymph";

            % Adding fat bodies from starved animals did not change the
            % concentration of Dilps in brain.
            "starved fat-bodies" ->[thin,dotted] release;

            % However adding fat bodies from well fed animal reduce the
            % concentration of Dilps in brain;
            "fed fat-bodies" ->[-*] release;

            "depolarization/KCl" [blue] ->[-*] release;
        };
    \end{tikzpicture}    
\end{wrapfigure}


\section{Summary}

\begin{wrapfigure}{l}{0.4\textwidth}
    \centering
    \begin{tikzpicture}[scale=1 , every node/.style={} ]
        \graph [ edges = { thick }, nodes = {}, layered layout ]
        {
            "Insulin Producing Cells" -> secretion[blue] -> "Dilps/IPC"
                -> release[blue] -> "Dilps/hemolymph";

            feeding[blue] -> nutrients;
            nutrients -> "fat body" -> "FactorX/hemolymph" -> "???"[blue] -> release;

            "FactorX/hemolymph" ->[dotted,thin, "??"] "hyperpolarization";

            secretion[blue];


            "hyperpolarization" [blue] ->[-|] "release";

            secretion [ blue] -> "Dilps/IPC" -> "release" -> "Dilps/hemolymph"
            -> growth [blue];

            TOR ->[-*] "release";
        };
    \end{tikzpicture}    
\end{wrapfigure}

\end{document}
